### PR TITLE
ci: matrix ci for k8s

### PR DESCRIPTION
**Pull request by liangyuanpeng**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

1. CI support matrix with k8s version
2. Upload e2e ci log to github action artifact
3. concurrency config for e2e ci

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -33,4 +39,15 @@ jobs:
...
         run: bash ./tests/e2e/e2e.sh -p ci
+      - name: clean
+        if: failure()
+        run: bash ./tests/e2e/e2e.sh -c
+      - name: upload kind logs
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: xline_kind_log_${{ matrix.k8s }}
+          path: /tmp/xlineoperator/e2e-kind
```
Comment (iGxnon) [inline .github/workflows/e2e.yml:53]: a new line at the end of file

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -14,6 +17,9 @@ jobs:
   validation:
     name: 'Validation'
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        k8s: [ v1.25.3,v1.26.0, v1.27.3]
```
Comment (iGxnon) [inline .github/workflows/e2e.yml:22]: add a space here? `[ v1.25.3, v1.26.0, v1.27.3 ]`
BTW, should we support these k8s versions even we update operator?

**Review by iGxnon — COMMENTED** [review]
Thanks for your contribution! We are looking forward to receiving your responses to the comments within this PR.

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (liangyuanpeng) [reply .github/workflows/e2e.yml:22]: Once the operator uses the API of the new version of K8S API, the minimum version of the CI should be replace to this version.

Basically, CI can run with kubernetes latest stable version -2/-3.

Comment (liangyuanpeng) [reply .github/workflows/e2e.yml:22]: Can run with schedule for more k8s version, if it's necessarily.

**Review by iGxnon — DISMISSED** [review]
LGTM

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -5,6 +5,9 @@ on:
     branches:
       - main
   workflow_dispatch: { }
+concurrency:
```
Comment (Phoenix500526) [inline .github/workflows/e2e.yml:8]: Please add a blank line before this line for better readability.

Comment (liangyuanpeng) [reply .github/workflows/e2e.yml:8]: IMO,a blank line is unnecessarily, so just Add the new line with doc.

Comment (liangyuanpeng) [reply .github/workflows/e2e.yml:10]: Added new line

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by iGxnon — APPROVED** [review]

**Pull request merged**